shell.nix

```diff
@@ -0,0 +1,11 @@
+{
```

Comment:
Is this part of the PR or an accidental inclusion?

Reply:
Not accidental, but probably belong into another PR: if you don't mind I would like to add this file and wire it up from `flake.nix`. With a `shell.nix`, and maybe an `.envrc`, when a Nix user cd into the project, it sources everything needed to work on the project (that is even if they don't have golang installed on their machine).

Reply:
Sure, we can keep it.